src/graphql/event/dns.rs

```diff
@@ -144,6 +149,26 @@ impl DnsCovertChannel {
             .as_ref()
             .map(|scores| scores.iter().map(Into::into).collect::<Vec<TriageScore>>())
     }
+
+    async fn level(&self) -> ThreatLevel {
+        Self::get_threat_level()
+    }
+
+    async fn learning_method(&self) -> LearningMethod {
+        Self::get_learning_method()
```

Comment:
The same goes here; removing the `EventLearningMethod` trait, it would be better to return `LearningMethod::SemiSupervised` directly.